examples/xlio_ultra_api_ping_pong.c

```diff
@@ -822,115 +697,100 @@ static void socket_accept_cb(xlio_socket_t sock, xlio_socket_t parent, uintptr_t
-     * ================================================
-     */
-    struct xlio_socket_send_attr send_attr = {
-        .flags = XLIO_SOCKET_SEND_FLAG_FLUSH, /* Flush immediately */
-        .mkey = g_state.mr_buf->lkey, /* Memory key for zero-copy */
-        .userdata_op = (uintptr_t)msg /* User data for completion callback */
-    };
+    /* Step 3: Process all complete messages in the accumulator */
+    msg_start = acc->data;
+    while (acc->len > 0) {
+        /* Search for the message delimiter in current accumulated data */
+        newline_pos = memchr(msg_start, '\n', acc->len);
+        if (!newline_pos) {
+            break;
+        }
 
-    /*
-     * Send data using XLIO Ultra API
-     * ==============================
-     */
-    int rc = g_state.api->xlio_socket_send(sock, g_state.sndbuf, len, &send_attr);
-    if (rc != 0) {
-        fprintf(stderr, "Failed to send message: %s\n", strerror(errno));
-        return rc;
-    }
+        /* Found a complete message */
+        msg_len = newline_pos - msg_start;
+        /* Make a null-terminate message in-place */
+        *newline_pos = '\0';
 
-    return 0;
-}
+        printf("Received message: '%s' (%zu bytes)\n", msg_start, msg_len);
 
-/*
- * Send ping message with RTT timing
- * =================================
- * Records timestamp when sending ping for RTT measurement.
- */
-static int send_ping_with_timing(xlio_socket_t sock)
-{
-    /* Record timestamp before sending ping */
-    if (clock_gettime(CLOCK_MONOTONIC, &g_state.ping_sent_time) != 0) {
-        fprintf(stderr, "Failed to get timestamp: %s\n", strerror(errno));
-        return -1;
+        /* Process the null-terminated message */
+        process_received_message(sock, msg_start, msg_len);
+
+        /* Move to the next message */
+        msg_start = newline_pos + 1;
+        remaining = acc->len - (msg_start - acc->data);
+        acc->len = remaining;
     }
 
-    /* Send the ping message */
-    return send_message(sock, MSG_PING);
+    /* Step 4: Move any remaining incomplete message to start of buffer */
+    if (acc->len > 0 && msg_start != acc->data) {
+        memmove(acc->data, msg_start, acc->len);
+    }
 }
 
 /*
- * Calculate RTT in microseconds
- * =============================
- * Calculate round-trip time between two timespec timestamps.
- */
-static double calculate_rtt_microseconds(const struct timespec *start, const struct timespec *end)
-{
-    long long start_ns = start->tv_sec * 1000000000LL + start->tv_nsec;
-    long long end_ns = end->tv_sec * 1000000000LL + end->tv_nsec;
-    long long diff_ns = end_ns - start_ns;
-
-    return diff_ns / 1000.0;
-}
-
-/*
- * Process received message based on application logic
- * ===================================================
+ * Process received message based on application logic
+ * ===================================================
+ * The function is common for server and client and reacts on the message value:
```

Comment:
The function is common for server and client and reacts on the message value: -> 
The function is common for server and client and reacts to the message value: